src/openforms/submissions/views.py

```diff
@@ -110,29 +114,11 @@ def is_auth_data_correct(self, submission: Submission) -> bool:
 
         if not meets_plugin_requirements(self.request, submission.form, plugin_id):
             return False
-        is_auth_attribute_correct = (
-            submission.auth_info.attribute
-            == self.request.session[FORM_AUTH_SESSION_KEY]["attribute"]
-        )
 
-        submission_auth_value = submission.auth_info.value
-        # there are two modus operandi - the submission may not be completed yet, in
-        # which case we don't have a hashed value yet, but the raw value (used for
-        # prefill and the like). There are also other flows where the attributes may
-        # be hashed despite the submission not being completed yet.
-        current_auth_value = self.request.session[FORM_AUTH_SESSION_KEY]["value"]
-
-        if submission.auth_info.attribute_hashed:
-            is_auth_data_correct = check_salted_hash(
-                current_auth_value, submission_auth_value, setter=None
-            )
-        else:
-            # timing attacks are not a concern here, as you need to go through a valid
-            # authentication flow first which sets the value in the session. Additionally,
-            # enumeration would still be possible and you don't necessarily leak passwords
-            # that may be used on other sites - end-users cannot change their BSN/KVK etc.
-            # anyway.
-            is_auth_data_correct = submission_auth_value == current_auth_value
+        auth_attribute = self.request.session[FORM_AUTH_SESSION_KEY]["attribute"]
+        auth_value = self.request.session[FORM_AUTH_SESSION_KEY]["value"]
+        is_auth_attribute_correct = is_same_login_attribute(submission, auth_attribute)
+        is_auth_data_correct = is_same_login_value(submission, auth_value)
```

Comment:
I don't think these need to be separate functions, ultimately what we want to know is:

```py
is_same_user = check_user_matches_submission(self.request, submission)
```
with `check_user_matches_submission` imported from `openforms.authentication.service`

which turns this methods return statement into:

```py
return is_auth_plugin_correct and is_same_user
```